keps/sig-network/3070-reserved-service-ip-range/README.md

```diff
@@ -0,0 +1,436 @@
+lower range. This will allow users to define static allocations on the lower band of the predefined
+range with a low risk of collision, keeping the implementation backwards compatible.
+
+Example 1 Large Subnet:
+
+- Service IP CIDR: 192.168.0.0/16
+- Range Size: 2^16 - 2 = 65534
+- Band Offset: min(max(16,65536/16),256) = min(4096,256) = 256
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.1.1
+
+           ┌─────────────┬─────────────────────────────────────────────┐
+           │   static    │                    dynamic                  │
+           └─────────────┴─────────────────────────────────────────────┘
+
+           ◄────────────► ◄────────────────────────────────────────────►
+    192.168.0.1     192.168.1.1                                 192.168.255.254
+
+Example 2:
+
+- Service IP CIDR: 192.168.0.0/22
+- Range Size: 2^10 - 2 = 1022
+- Band Offset: min(max(16,1024/16),256) = min(max(16,64),256) = 64
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.0.64
+
+           ┌─────────────┬─────────────────────────────────────────────┐
+           │   static    │                    dynamic                  │
+           └─────────────┴─────────────────────────────────────────────┘
+
+           ◄────────────► ◄────────────────────────────────────────────►
+    192.168.0.1     192.168.0.64                                 192.168.3.254
+
+Example 3 Small Subnet:
+
+- Service IP CIDR: 192.168.0.0/26
+- Range Size: 2^6 - 2 = 62
+- Band Offset: min(max(16,64/16),256) = min(max(16,4),256) = 16
+- Static band start: 192.168.0.1
+- Static band ends: 192.168.0.16
+
+           ┌─────────────┬─────────────────────────────────────────────┐
+           │   static    │                    dynamic                  │
+           └─────────────┴─────────────────────────────────────────────┘
+
+           ◄────────────► ◄────────────────────────────────────────────►
+    192.168.0.1     192.168.0.16                                 192.168.0.62
+
+
+```go
+// randomScanReservedStrategy choose a random address from the provided big.Int and then scans
+// forward looking for the next available address. The big.Int range is subdivided so it will try
+// to allocate first from the reserved block of addresses (it will wrap the reserved subrange if necessary).
+// If there is no free address it will try to allocate one from the reserved block too.
+type randomScanReservedStrategy struct {
+	rand     *rand.Rand
+	reserved int
+}
+```
+
+### Test Plan
+
+This feature doesn't modify the cluster behavior, only the order on which dynamic IP are assigned to Services,
+there is no need for e2e or integration tests, unit tests are enough.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+
+#### Beta
+
+- Gather feedback from developers and community.
+- No issues reported.
+
+#### GA
+
+- 2 examples of real-world usage
```

Comment:
What does this mean?  That 2 providers have switched to this mode as the default?  That's unlikely while it is beta.  Let's get specific here.  I think time and lack of trouble is the only real signal.

When do we make this mode the default?  Is there a reason not to do that?

Reply:
It is interest in enabling it in Openshift by default to solve some races at installation time, I was expecting more installers will face similar issue, but as you say this is just more a matter of time and signal.